pandas/core/indexing.py

```diff
@@ -2164,11 +2166,14 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
         result = result.astype(bool)._values
     elif is_object_dtype(key):
         # key might be object-dtype bool, check_array_indexer needs bool array
+
         result = np.asarray(result, dtype=bool)
         result = check_array_indexer(index, result)
     else:
+        if not is_array_like(result):
+            # GH 33924
+            result = pd_array(result, dtype=bool)
```

Comment:
Why not just np.array?

Reply:
because of nullable input

Reply:
if the input is nullable, wouldn't `not is_array_like(result)` be `False`?

Reply:
I meant that this array/list can have nullable elements.